qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,800 @@
+    # Wallet
+    ('5a', True, True), # After generating multiple z-addrs, run z_listaddresses
+    ('5b', True, True), # Run z_validateaddress with a Sprout z-addr
+    ('5c', True, True), # Run z_validateaddress with a Sapling z-addr
+    ('5d', True, True), # After a transaction, run z_listunspent
+    ('5e', True, True), # After a transaction, run z_listreceivedbyaddress
+    ('5f', True, True), # After a transaction, run z_getbalance
+    ('5g', True, True), # After a transaction, run z_gettotalbalance
+    ('5h', True, True), # Run z_exportkey using a Sprout z-addr
+    ('5i', True, True), # Run z_importkey using the zkey from a Sprout z-addr
+    ('5j', True, True), # Run z_exportkey using a Sapling z-addr
+    ('5k', True, True), # Run z_importkey using the zkey from a Sapling z-addr
+    ('5l', True, True), # Run z_exportwallet
+    ('5m', True, True), # Run z_importwallet
+    ('5n', True, True), # Run z_shieldcoinbase
+    ('5o', True, True), # Run getwalletinfo
+    # Network
+    ('6a', True, True), # Run getpeerinfo
+    ('6b', True, True), # Run getnetworkinfo
+    ('6c', True, False), # Run getdeprecationinfo
+    ('6d', True, True), # Run getconnectioncount
+    ('6e', True, True), # Run getaddednodeinfo
+    # Mining
+    ('7a', True, True), # Run getblocksubsidy
+    ('7b', True, True), # Run getblocktemplate
+    ('7c', True, True), # Run getmininginfo
+    ('7d', True, True), # Run getnetworkhashps
+    ('7e', True, True), # Run getnetworksolps
+]
+
+
+#
+# Test helpers
+#
+
+def run_cmd(results, case, zcash, name, args=[]):
+    print('-----')
+    print('%s $ zcash-cli %s %s' % (
+        case.ljust(3),
+        name,
+        ' '.join([str(arg) for arg in args],
+    )))
+    try:
+        res = zcash.__getattr__(name)(*args)
+        print(res)
+        print()
+        if results is not None and len(case) > 0:
+            results[case] = True
+        return res
+    except RpcException as e:
+        print('ERROR: %s' % str(e))
+        if results is not None and len(case) > 0:
+            results[case] = False
+        return None
+
+def wait_for_balance(zcash, zaddr, expected=None, timeout=900):
+    print('Waiting for funds to %s...' % zaddr)
+    if expected is not None:
+        print('Expecting %s' % expected)
+
+    unconfirmed_balance = zcash.z_getbalance(zaddr, 0)
+    print('Unconfirmed balance: %s' % unconfirmed_balance)
+    if expected is not None and unconfirmed_balance != expected:
+        print('WARNING: Unconfirmed balance does not match expected balance')
+
+    ttl = timeout
+    while True:
+        balance = zcash.z_getbalance(zaddr)
+        if balance == unconfirmed_balance:
+            print('Received %s' % balance)
+            return balance
+
+        time.sleep(1)
+        ttl -= 1
+        if ttl == 0:
+            # Ask user if they want to keep waiting
+            print()
```

Comment:
Maybe add a line here about what is going on before the user has to make a decision:
```
print('TIMEOUT: Received %s, Expected %s' % (balance, expected))
```